### DISCUSSION TITLE
Cannot execute actions - The passed action type is invalid.

### DISCUSSION THREAD

**Discussion by Pankaj2101** [iRobotEducation/create3_docs#317] [Questions] [answered]
I just got a iRobot Create 3, connected it to Wifi & installed ROS2 galactic on my PC. set the `RMW_IMPLEMENTATION=rmw_cyclonedds_cpp`
Tested  - PC is able to ping robot - `ros2 topic list' & 'ros2 action list` etc work well showing Create3 items.
Tested ros2 as well by running -  `ros2 run demo_nodes_cpp talker` & `ros2 run demo_nodes_py listener` in different terminals and they are working well.

However as soon as I execute `ros2 action send_goal /undock irobot_create_msgs/action/Undock "{}"`
I get the error `The passed action type is invalid`

I believe the messages package is/are not installed and so tried.
```
sudo apt-get update
sudo apt-get-install ros-galactic-irobot-create-msgs
```
but to no avail.

Can someone guide me to simple resolve.

**Comment by shamlian** [comment, marked as answer]
I would suggest to run the following command after installing the irobot create messages package
```
source /opt/ros/galactic/setup.sh
```

**Reply by Pankaj2101** [reply]
Did not work.
[My bad - I should have posted this in issues]

The OS is Ubuntu 20.04 LTS

**Reply by Pankaj2101** [reply]
++ when I install msgs thru
sudo apt-get update
sudo apt-get-install ros-galactic-irobot-create-msgs

I get a lot of warnings like 'W: Target Packages (restricted/binary-amd64/Packages) is configured multiple times in /etc/apt/spirces.list:42 and /etc/apt/sources.list:54'

**Reply by alsora** [reply]
@Pankaj2101 please post a full log of the output of the following.

```
sudo apt-get update
sudo apt-get-install ros-galactic-irobot-create-msgs
source /opt/ros/galactic/setup.sh
ros2 action send_goal /undock irobot_create_msgs/action/Undock "{}"
```

This is the correct procedure, so there has to be something that is failing while you do that.

**Reply by Pankaj2101** [reply]
[ExecutionLog.txt](https://github.com/iRobotEducation/create3_docs/files/10490947/ExecutionLog.txt)

Is there anywhere that logs are generated? or some directories/files created?

**Reply by Pankaj2101** [reply]
Worked after i ran the commands individually.
